app/models/delete_queue_job.rb

```diff
@@ -92,6 +88,18 @@ def perform
           study.default_options[:annotation] = nil
           study.save
         end
+      when 'AnnData'
+        delete_convention_data(study: study, metadata_file: object)
+        # delete user annotations first as we lose associations later
+        delete_user_annotations(study:, study_file: object)
+        delete_parsed_data(object.id, study.id, ClusterGroup, CellMetadatum, Gene, DataArray)
+        # reset default options/counts
+        study.reload
+        study.cell_count = study.all_cells_array.size
+        study.gene_count = study.unique_genes.size
+        study.default_options[:cluster] = nil
+        study.default_options[:annotation] = nil
```

Comment:
At a glance, this seems like it might only handle when the study has only 1 study file, an AnnData file?  E.g. if the user has a traditional set of metadata, clustering, and expression files _and_ an AnnData file, then presumably they could still have a non-nil default `cluster` and `annotation`.

Reply:
That is true - it does.  This was a conscious decision since we're currently only working on "happy path" for AnnData.  Currently, we would not be able to support a user uploading a traditional metadata file alongside an AnnData file.  We could support other clustering files though without any other updates, so I'll take a quick pass and making sure we wouldn't unset that if there was a default that didn't come from the AnnData file.

Reply:
[1a65dcf](https://github.com/broadinstitute/single_cell_portal_core/pull/1701/commits/1a65dcf0241545adeed54578f9645491fef3c3fc)